lib/src/reporters/slack_reporter.dart

```diff
@@ -0,0 +1,255 @@
+  final int maximumToleratedFailures;
+
+  final SlackMessenger slackMessenger;
+
+  /// Reported as the initial message (or parent thread when [threadResults] is `true`)
+  /// of the test run. Defaults to `Starting tests`.
+  final String startLabel;
+
+  /// Whether the test run should be threaded under a single message. Defaults `true`.
+  final bool threadResults;
+
+  final features = <FinishedMessage>[];
+  final scenarios = <ScenarioFinishedMessage>[];
+  final scenariosInActiveFeature = <ScenarioFinishedMessage>[];
+  StepFinishedMessage firstFailedStepInActiveScenario;
+
+  SlackReporter(
+    this.slackMessenger, {
+    this.maximumToleratedFailures = 15,
+    this.threadResults = true,
+    this.startLabel = 'Starting tests',
+  });
+
+  @override
+  Future<void> onException(exception, stackTrace) {
+    final payload = [
+      slackMessenger.divider,
+      slackMessenger.buildTextSection(':bangbang: Exception :bangbang:'),
+      slackMessenger.buildTextSection(exception.toString()),
+      slackMessenger.buildTextSection('```$stackTrace```'),
+      slackMessenger.divider,
+    ];
+
+    return slackMessenger.notifySlack(payload);
+  }
+
+  @override
+  Future<void> message(msg, level) async {
+    if (level == MessageLevel.error) await slackMessenger.sendText(msg);
+  }
+
+  @override
+  Future<void> onFeatureFinished(feature) async {
+    features.add(feature);
+    final allScenariosPassed = scenariosInActiveFeature.every((s) => s.passed);
+
+    if (allScenariosPassed) {
+      await slackMessenger.sendText(
+          ':white_check_mark: ${feature.name} (${scenariosInActiveFeature.length}/${scenariosInActiveFeature.length} passed)');
+    } else {
+      final passedScenarios = scenariosInActiveFeature.where((s) => s.passed);
+      await slackMessenger.sendText(
+          ':warning: ${feature.name} (${passedScenarios.length}/${scenariosInActiveFeature.length} passed)');
+    }
+
+    scenariosInActiveFeature.clear();
+  }
+
+  @override
+  Future<void> onScenarioFinished(scenario) async {
+    scenarios.add(scenario);
+    scenariosInActiveFeature.add(scenario);
+    if (!scenario.passed) {
+      final payload = [
+        slackMessenger.buildTextSection(':x: ${scenario.name}'),
+        slackMessenger.buildTextSection(
+            'Failed at step: ${firstFailedStepInActiveScenario.name}'),
+      ];
+      await slackMessenger
+          .notifySlack(payload)
+          .then((_) => firstFailedStepInActiveScenario = null);
+    }
+
+    if (scenarios.where((s) => !s.passed).length > maximumToleratedFailures) {
+      await slackMessenger.sendText(
+          ':x: :x: :x: :x: Aborting: too many failures :x: :x: :x: :x:');
+      exit(1);
+    }
+  }
+
+  @override
+  Future<void> onStepFinished(step) async {
+    if (step.result.result != StepExecutionResult.pass &&
+        firstFailedStepInActiveScenario == null) {
+      firstFailedStepInActiveScenario = step;
+      for (var attachment in step.attachments) {
```

Comment:
@jonsamwell I added this here to [address your feedback](https://github.com/jonsamwell/flutter_gherkin/pull/71#discussion_r456848824). As this repo doesn't access `FlutterWorld`, it didn't make sense to include the hook that generates the screenshot. I [documented it](https://github.com/jonsamwell/dart_gherkin/pull/13/files#diff-65bc21c9687167a834875707826f5aadR118-R130) for the implementor's convenience. Do you think that example is sufficient or should it be more visible/cross-documented?